geopandas/io/tests/test_arrow.py

```diff
@@ -578,3 +626,83 @@ def test_write_read_feather_expand_user():
     f_df = geopandas.read_feather(test_file)
     assert_geodataframe_equal(gdf, f_df, check_crs=True)
     os.remove(os.path.expanduser(test_file))
+
+
+@pytest.mark.parametrize("format", ["feather", "parquet"])
+def test_write_read_default_crs(tmpdir, format):
+    if format == "feather":
+        from pyarrow.feather import write_feather as write
+    else:
+        from pyarrow.parquet import write_table as write
+
+    filename = os.path.join(str(tmpdir), f"test.{format}")
+    gdf = geopandas.GeoDataFrame(geometry=[box(0, 0, 10, 10)])
+    table = _geopandas_to_arrow(gdf)
+
+    # update the geo metadata to strip 'crs' entry
+    metadata = table.schema.metadata
+    geo_metadata = _decode_metadata(metadata[b"geo"])
+    del geo_metadata["columns"]["geometry"]["crs"]
+    metadata.update({b"geo": _encode_metadata(geo_metadata)})
+    table = table.replace_schema_metadata(metadata)
+
+    write(table, filename)
+
+    read = getattr(geopandas, f"read_{format}")
+    df = read(filename)
+    assert df.crs.equals(CRS("OGC:CRS84"))
+
+
+@pytest.mark.parametrize(
+    "format,version", product(["feather", "parquet"], [None] + SUPPORTED_VERSIONS)
+)
+def test_write_spec_version(tmpdir, format, version):
+    if format == "feather":
+        from pyarrow.feather import read_table
+
+    else:
+        from pyarrow.parquet import read_table
+
+    filename = os.path.join(str(tmpdir), f"test.{format}")
+    gdf = geopandas.GeoDataFrame(geometry=[box(0, 0, 10, 10)], crs="EPSG:4326")
+    write = getattr(gdf, f"to_{format}")
+    write(filename, version=version)
+
+    # ensure that we can roundtrip data regardless of version
+    read = getattr(geopandas, f"read_{format}")
+    df = read(filename)
+    assert_geodataframe_equal(df, gdf)
+
+    table = read_table(filename)
+    metadata = json.loads(table.schema.metadata[b"geo"])
+    assert metadata["version"] == version or METADATA_VERSION
+
+    # verify that CRS is correctly handled between versions
+    if version == "0.1.0":
+        assert metadata["columns"]["geometry"]["crs"] == gdf.crs.to_wkt()
+
+    else:
+        assert metadata["columns"]["geometry"]["crs"] == gdf.crs.to_json_dict()
+
+
+@pytest.mark.parametrize("version", ["0.1.0", "0.4.0"])
+def test_read_versioned_file(version):
+    # Verify that files for different metadata spec versions can be read
+    # created for each supported version:
+    # df = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
+    # df.head(2).to_feather(DATA_PATH / 'arrow' / f'naturalearth_lowres_top2_v{METADATA_VERSION}.feather')  # noqa: E501
+    # df.head(2).to_parquet(DATA_PATH / 'arrow' / f'naturalearth_lowres_top2_v{METADATA_VERSION}.parquet')  # noqa: E501
+
+    check_crs = Version(pyproj.__version__) >= Version("3.0.0")
```

Comment:
This is a bit of a dirty trick, as the crs objects were not perfectly comparable according to some of our pyproj 2.6 CI tests; I don't know what version of pyproj specifically made sure that these are more comparable or even if the comparison holds up across a larger range of versions.  The files were written with pyproj 3.2.1, so it appears to have worked to limit this check to >=3.0

Reply:
This is fine I think, I will need to same in https://github.com/geopandas/geopandas/pull/2443